rdr_service/genomic/genomic_storage_class.py

```diff
@@ -0,0 +1,130 @@
+
+
+class GenomicStorageClass:
+
+    def __init__(self,
+                 storage_job_type,
+                 storage_class='COLDLINE'
+                 ):
+        self.storage_job_type = storage_job_type
+        self.storage_class = storage_class
+        self.updated_count = 0
+
+        self.storage_provider = GoogleCloudStorageProvider()
+        self.metrics_dao = GenomicGCValidationMetricsDao()
+        self.storage_update_dao = GenomicDefaultBaseDao(
+            model_type=GenomicStorageUpdate
+        )
+
+    def __get_storage_method(self):
+        return {
+            GenomicJob.UPDATE_ARRAY_STORAGE_CLASS: self.update_array_files,
+            GenomicJob.UPDATE_WGS_STORAGE_CLASS: self.update_wgs_files
+        }[self.storage_job_type]
+
+    def run_storage_update(self):
+        run_method = self.__get_storage_method()
+        run_method()
+
+    @classmethod
+    def get_file_dict_from_metrics(cls, *,
+                                   metrics: List[GenomicGCValidationMetrics],
+                                   metric_type: str
+                                   ) -> List[Dict[str, Union[int, list]]]:
+
+        metric_file_type_map = {
+            config.GENOME_TYPE_ARRAY: [obj['file_path_attribute'] for obj in array_file_types_attributes],
+            config.GENOME_TYPE_WGS: [obj['file_path_attribute'] for obj in wgs_file_types_attributes]
+        }[metric_type]
+
+        files_to_update = []
+        for metric in metrics:
+            metric_obj = {'metric_id': metric.id,
+                          'metric_paths': [obj[1] for obj in metric if 'Path' in obj[0] and obj[0] in
+                                           metric_file_type_map and obj[1]
+                                           is not None],
+                          'metric_type': metric_type}
+
+            files_to_update.append(
+                metric_obj
+            )
+        return files_to_update
+
+    def update_array_files(self):
+        array_metrics = self.metrics_dao.get_fully_processed_metrics()
+
+        if not array_metrics:
+            logging.info('There are currently no array data files to update')
+            return
+
+        self.update_file_paths(
+            file_dict=self.get_file_dict_from_metrics(
+                metrics=array_metrics,
+                metric_type=config.GENOME_TYPE_ARRAY
+            )
+        )
+
+    def update_wgs_files(self):
+        wgs_metrics = self.metrics_dao.get_fully_processed_metrics(
+            metric_type=config.GENOME_TYPE_WGS
+        )
+
+        if not wgs_metrics:
+            logging.info('There are currently no wgs data files to update')
+            return
+
+        self.update_file_paths(
+            file_dict=self.get_file_dict_from_metrics(
+                metrics=wgs_metrics,
+                metric_type=config.GENOME_TYPE_WGS
+            )
+        )
+
+    def update_file_paths(self, file_dict: List[dict]):
```

Comment:
Nitpicky, but can you name this something like "reclassify" or "update_storage_class_for_file_paths"? `update_file_paths` makes me think that we're changing the file paths themselves, not the storage class.